Emby.Server.Implementations/SyncPlay/GroupController.cs

```diff
@@ -0,0 +1,662 @@
+                    Ping = DefaultPing,
+                    IsBuffering = false
+                });
+        }
+
+        /// <summary>
+        /// Removes the session from the group.
+        /// </summary>
+        /// <param name="session">The session.</param>
+        private void RemoveSession(SessionInfo session)
+        {
+            Participants.Remove(session.Id);
+        }
+
+        /// <summary>
+        /// Filters sessions of this group.
+        /// </summary>
+        /// <param name="from">The current session.</param>
+        /// <param name="type">The filtering type.</param>
+        /// <returns>The list of sessions matching the filter.</returns>
+        private IEnumerable<SessionInfo> FilterSessions(SessionInfo from, SyncPlayBroadcastType type)
+        {
+            return type switch
+            {
+                SyncPlayBroadcastType.CurrentSession => new SessionInfo[] { from },
+                SyncPlayBroadcastType.AllGroup => Participants
+                    .Values
+                    .Select(session => session.Session),
+                SyncPlayBroadcastType.AllExceptCurrentSession => Participants
+                    .Values
+                    .Select(session => session.Session)
+                    .Where(session => !session.Id.Equals(from.Id, StringComparison.OrdinalIgnoreCase)),
+                SyncPlayBroadcastType.AllReady => Participants
+                    .Values
+                    .Where(session => !session.IsBuffering)
+                    .Select(session => session.Session),
+                _ => Enumerable.Empty<SessionInfo>()
+            };
+        }
+
+        /// <summary>
+        /// Checks if a given user can access a given item, that is, the user has access to a folder where the item is stored.
+        /// </summary>
+        /// <param name="user">The user.</param>
+        /// <param name="item">The item.</param>
+        /// <returns><c>true</c> if the user can access the item, <c>false</c> otherwise.</returns>
+        private bool HasAccessToItem(User user, BaseItem item)
+        {
+            var collections = _libraryManager.GetCollectionFolders(item)
+                .Select(folder => folder.Id.ToString("N", CultureInfo.InvariantCulture));
+            return collections.Intersect(user.GetPreference(PreferenceKind.EnabledFolders)).Any();
+        }
+
+        /// <summary>
+        /// Checks if a given user can access all items of a given queue, that is,
+        /// the user has the required minimum parental access and has access to all required folders.
+        /// </summary>
+        /// <param name="user">The user.</param>
+        /// <param name="queue">The queue.</param>
+        /// <returns><c>true</c> if the user can access all the items in the queue, <c>false</c> otherwise.</returns>
+        private bool HasAccessToQueue(User user, IReadOnlyList<Guid> queue)
+        {
+            // Check if queue is empty.
+            if (!queue?.Any() ?? true)
```

Comment:
```suggestion
            if (queue == null || queue.Count == 0)
```

Reply:
I'm sorry to be bothering you with these dumb one liners but Roslynator suggested that. That's also why I added that extra comment hahaha, I was like: "If this is the C# way of doing things then be it.". The suggestion pops up only for `IEnumerable`. :man_shrugging:

Reply:
Don't know why but the analyzers seem to prefer Linq / IEnumerable.... We're just trying to avoid it unless it makes the code much simpler :)